src/crawlee/_autoscaling/_types.py

```diff
@@ -103,6 +109,11 @@ class MemorySnapshot:
     @property
     def is_overloaded(self) -> bool:
         """Indicate whether the memory is considered as overloaded."""
+        if self.system_wide_memory_size is not None and self.system_wide_used_size is not None:
+            system_wide_utilization = self.system_wide_used_size / self.system_wide_memory_size
+            if system_wide_utilization > 0.97:
```

Comment:
It would be tricky to make this configurable. I guess I could make it a constant though.